### PR TITLE
Overmap sensors overhaul

### PR DESCRIPTION
**Pull request by FluffyGhoster**
The ship sensors now work as a sonar or radar, obscuring the view when the first object is pinged, otherwise the things not in range will not be seen.

The ships now need to be scanned before they appear and are identified, which takes a little of time.

### Asset Licenses
The following assets that **have not** been created by myself are included in this PR:

Everything added comes from Nebula, which is released as far as I'm aware under the Affero license

### CODE REVIEW

File: code/modules/overmap/ships/computers/sensors.dm [hunk 1]
```diff
@@ -336,6 +371,11 @@
 	take_damage(20/severity)
 	toggle()
 
+// /obj/machinery/shipsensors/RefreshParts()
+// 	..()
+// 	//sensor_strength = clamp(total_component_rating_of_type(/obj/item/stock_parts/manipulator), 0, 5)
+// 	sensor_strength = 5
+
```
Comment (NonQueueingMatt) [inline code/modules/overmap/ships/computers/sensors.dm:378]: commented code

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/modules/overmap/contacts/_contacts.dm [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	if(length(showing))
+		for(var/datum/weakref/W in showing)
+			var/mob/M = W.resolve()
+			if(istype(M) && M.client)
+				M.client.images |= images
+
+/// This function checks if the shield is active, as we likely will need to rework the shields to be anything useful and gameplay-wise engaging this is currently not used, but left here for when
+/// it will happen
+
+// /datum/overmap_contact/proc/check_effect_shield()
```
Comment (NonQueueingMatt) [inline code/modules/overmap/contacts/_contacts.dm:75]: commented code

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

**Comment by FluffyGhoster** [comment]
!review

File: code/modules/overmap/sectors.dm [hunk 3]
```diff
@@ -41,6 +41,33 @@ var/global/area/overmap/map_overmap // Global object used to locate the overmap
...
+	base_type = base_type || machine.type
+	if(islist(associated_machinery) && associated_machinery[base_type])
+		LAZYREMOVE(associated_machinery[base_type], WEAKREF(machine))
+
+/obj/effect/overmap/visitable/proc/register_machine(var/obj/machinery/machine, var/base_type)
+	ASSERT(istype(machine))
+	if(!QDELETED(machine))
+		base_type = base_type || machine.type
+		LAZYINITLIST(associated_machinery)
+		LAZYDISTINCTADD(associated_machinery[base_type], WEAKREF(machine))
```
Comment (NonQueueingMatt) [inline code/modules/overmap/sectors.dm:69]: All of this is assorted machinery stuff is unused now, so get rid of it

File: code/modules/overmap/contacts/tracker.dm [hunk 4]
```diff
@@ -0,0 +1,24 @@
+/obj/item/ship_tracker
+	name = "long range tracker"
+	desc = "A complex device that transmits conspicuous signals, easily locked onto by modern sensors hardware."
+	// icon = 'icons/obj/ship_tracker.dmi'
+	// icon_state = "disabled"
```
Comment (NonQueueingMatt) [inline code/modules/overmap/contacts/tracker.dm:5]: Port the icon for this

File: code/modules/overmap/contacts/tracker.dm [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+	origin_tech = "{'magnets':3, 'programming':2}"
+	var/enabled = FALSE
+
+/obj/item/ship_tracker/Initialize()
+	. = ..()
+	// set_extension(src, /datum/extension/interactive/multitool/store)
```
Comment (NonQueueingMatt) [inline code/modules/overmap/contacts/tracker.dm:12]: commented code

File: code/modules/overmap/sectors.dm [hunk 6]
```diff
@@ -41,6 +41,33 @@ var/global/area/overmap/map_overmap // Global object used to locate the overmap
 	/// Whether away ship comms have access to the common channel / PUB_FREQ
 	var/use_common = FALSE
 
+	var/list/associated_machinery
```
Comment (NonQueueingMatt) [inline code/modules/overmap/sectors.dm:44]: remove this too

File: code/modules/overmap/ships/computers/ship.dm [hunk 7]
```diff
@@ -102,6 +110,9 @@ somewhere on that shuttle. Subtypes of these can be then used to perform ship ov
 		return 0
 
 /obj/machinery/computer/ship/Destroy()
+	if(linked)
+		linked.unregister_machine(src, /obj/machinery/computer/ship)
+		linked = null
```
Comment (NonQueueingMatt) [inline code/modules/overmap/ships/computers/ship.dm:115]: this bit needs to be removed since unregister_machine doesn't work

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Review by Alberyk — APPROVED** [review]

**Pull request merged**